### PR TITLE
Sync with master branch

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: docs/guide/ru/README.md
Comment (vjik) [inline docs/guide/ru/README.md:1]: Add link to this file from readme in root

File: docs/guide/ru/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+### Валидация
+
+- [Использование валидатора](using-validator.md)
+- [Результат](result.md)
+- [Условия валидации](conditional-validation.md)
+- [Настройка сообщений об ошибках](customizing-error-messages.md)
+
+### Rules
+
+- [Built-in rules](built-in-rules.md)
```
Comment (vjik) [inline docs/guide/ru/README.md:16]: Broken link. Check other links also

File: messages/de/yii-validator.php
Comment (vjik) [inline messages/de/yii-validator.php:1]: Add line to changelog

File: src/Exception/UnexpectedRuleException.php [hunk 2]
```diff
@@ -15,11 +15,12 @@
  * use Yiisoft\Validator\Exception\UnexpectedRuleException;
  * use Yiisoft\Validator\Result;
  * use Yiisoft\Validator\RuleHandlerInterface;
+ * use Yiisoft\Validator\RuleInterface;
  * use Yiisoft\Validator\ValidationContext;
  *
  * final class MyRuleHandler implements RuleHandlerInterface
  * {
- *     public function validate(mixed $value, object $rule, ValidationContext $context): Result
+ *     public function validate(mixed $value, RuleInterface $rule, ValidationContext $context): Result
```
Comment (vjik) [inline src/Exception/UnexpectedRuleException.php:23]: In this version need `object`.

File: src/Rule/AtLeast.php [hunk 3]
```diff
@@ -65,6 +68,9 @@ public function __construct(
         private bool $skipOnError = false,
         private Closure|null $when = null
     ) {
+        if ($min > count($this->attributes)) {
```
Comment (vjik) [inline src/Rule/AtLeast.php:71]: Add line to changelog

File: src/Rule/EachHandler.php [hunk 4]
```diff
@@ -35,6 +35,8 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
         $rules = $rule->getRules();
         $result = new Result();
 
+        $originalEachKey = $context->getParameter(Each::PARAMETER_EACH_KEY);
```
Comment (vjik) [inline src/Rule/EachHandler.php:38]: Add line to changelog

**Review by vjik — CHANGES_REQUESTED** [review]

**Comment by samdark** [comment]
I don't think we need to backport changes from next major version into 1.x.

**Pull request closed without merging**